clients/pkg/logentry/stages/match.go

```diff
@@ -114,11 +114,23 @@ func newMatcherStage(logger log.Logger, jobName *string, config interface{}, reg
 }
 
 func getDropCountMetric(registerer prometheus.Registerer) *prometheus.CounterVec {
+	return getLogEntryMetric(registerer, "dropped_lines_total",
+		"A count of all log lines dropped as a result of a pipeline stage",
+		[]string{"reason"})
+}
+
+func getDropCountByLabelMetric(registerer prometheus.Registerer) *prometheus.CounterVec {
```

Comment:
done, moved this function to `limit.go`. Change to `match.go` was still required, because `limit.go` is  using `getDropCountMetric` from it and that function was refactored to become a wrapper around new `getLogEntryMetric`